dGame/Entity.cpp

```diff
@@ -1992,25 +1986,25 @@ void Entity::SetNetworkId(const uint16_t id) {
 	m_NetworkID = id;
 }
 
-std::vector<LWOOBJID>& Entity::GetTargetsInPhantom() {
-	std::vector<LWOOBJID> valid;
-
+std::vector<LWOOBJID> Entity::GetTargetsInPhantom() {
 	// Clean up invalid targets, like disconnected players
-	for (auto i = 0u; i < m_TargetsInPhantom.size(); ++i) {
-		const auto id = m_TargetsInPhantom.at(i);
-
-		auto* entity = Game::entityManager->GetEntity(id);
-
-		if (entity == nullptr) {
+	for (auto i = 0u; i < m_TargetsInPhantom.size(); ) {
+		if (Game::entityManager->GetEntity(m_TargetsInPhantom.at(i))) {
+			i++;
 			continue;
 		}
-
-		valid.push_back(id);
+		m_TargetsInPhantom.erase(m_TargetsInPhantom.begin() + i);
```

Comment:
This seems a weird way to filter a list, especially since there is another loop below that pushes to enemies?

Reply:
Well the first one checks for stale entities and prunes them, then a second loop actually does the checks for what to return.  I guess I could consolidate the loops but even then erase in a loop like this doesn't have high cost since there is no re allocation.  Best save here would just be using erase_if first and then copy_if after the null entities have been erased, but this is still 2 loops